src/include/gcc.h

```diff
@@ -125,21 +139,37 @@ WT_ATOMIC_CAS_FUNC(size, size_t *vp, size_t old, size_t newv)
 static inline bool
 __wt_atomic_cas_ptr(void *vp, void *old, void *newv)
 {
-    return (WT_ATOMIC_CAS((void **)vp, &old, newv));
+    return (WT_ATOMIC_CAS((void **)vp, &old, newv, WT_ATOMIC_SEQ_CST));
+}
+
+#define WT_ATOMIC_LOAD_PTR(type, vp, memorder) (type *)WT_ATOMIC_LOAD((void **)vp, memorder)
+
+/*
+ * __wt_atomic_memorder_cas_ptr --
+ *     Pointer compare and swap with memory ordering.
+ */
+static inline bool
+__wt_atomic_memorder_cas_ptr(void *vp, void *old, void *newv, int memorder)
+{
+    return (WT_ATOMIC_CAS((void **)vp, &old, newv, memorder));
```

Comment:
I think this is because it is an implicit cast from any pointer type to void* so we declare it as void * and then cast it to void **.

Reply:
Yes, it is legal, but seems fishy. For example, it would allow this incorrect code:

```c
int* atomic;
int* old = atomic;
int* new = malloc(4); // pretend I initialized the memory
__wt_atomic_memorder_cas_ptr(atomic, old, new, release); // oops! should have used &atomic
```